debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/AbstractRecordsProducerTest.java

```diff
@@ -126,54 +124,53 @@
                              new SchemaAndValueField("ss", Schema.INT16_SCHEMA, (short) 1),
                              new SchemaAndValueField("bs", Schema.INT64_SCHEMA, 123L),
-                             new SchemaAndValueField("b", Schema.OPTIONAL_BOOLEAN_SCHEMA, Boolean.TRUE));
-    }
-
-    protected List<SchemaAndValueField> schemasAndValuesForFpTypeWithoutDecimals() {
-        return Arrays.asList(new SchemaAndValueField("si", SchemaBuilder.OPTIONAL_INT16_SCHEMA, (short) 1),
-                             new SchemaAndValueField("i", SchemaBuilder.OPTIONAL_INT32_SCHEMA, 123456),
-                             new SchemaAndValueField("bi", SchemaBuilder.OPTIONAL_INT64_SCHEMA, 1234567890123L),
-                             new SchemaAndValueField("r", Schema.OPTIONAL_FLOAT32_SCHEMA, 3.0f),
-                             new SchemaAndValueField("db", Schema.OPTIONAL_FLOAT64_SCHEMA, 4.0d),
-                             new SchemaAndValueField("ss", Schema.INT16_SCHEMA, (short) 1),
-                             new SchemaAndValueField("bs", Schema.INT64_SCHEMA, 123L),
-                             new SchemaAndValueField("b", Schema.OPTIONAL_BOOLEAN_SCHEMA, Boolean.TRUE));
+                             new SchemaAndValueField("b", Schema.OPTIONAL_BOOLEAN_SCHEMA, Boolean.TRUE))
+                );
+        if (!DecoderDifferences.areSpecialFPValuesUnupported()) {
+            fields.addAll(Arrays.asList(
+                    new SchemaAndValueField("r_nan", Schema.OPTIONAL_FLOAT32_SCHEMA, Float.NaN),
+                    new SchemaAndValueField("db_nan", Schema.OPTIONAL_FLOAT64_SCHEMA, Double.NaN),
+                    new SchemaAndValueField("r_inf", Schema.OPTIONAL_FLOAT32_SCHEMA, Float.POSITIVE_INFINITY),
+                    new SchemaAndValueField("db_inf", Schema.OPTIONAL_FLOAT64_SCHEMA, Double.NEGATIVE_INFINITY)
+            ));
+        }
+        return fields;
     }
 
     protected List<SchemaAndValueField> schemasAndValuesForNumericDecimalType() {
         final Struct dvs = new Struct(VariableScaleDecimal.schema());
         dvs.put("scale", 4).put("value", new BigDecimal("10.1111").unscaledValue().toByteArray());
         final Struct nvs = new Struct(VariableScaleDecimal.schema());
         nvs.put("scale", 4).put("value", new BigDecimal("22.2222").unscaledValue().toByteArray());
+        final Struct dvs_nodec = new Struct(VariableScaleDecimal.schema());
+        dvs_nodec.put("scale", 0).put("value", new BigDecimal("10").unscaledValue().toByteArray());
+        final Struct nvs_nodec = new Struct(VariableScaleDecimal.schema());
+        nvs_nodec.put("scale", 0).put("value", new BigDecimal("22").unscaledValue().toByteArray());
         return Arrays.asList(
                 new SchemaAndValueField("d", Decimal.builder(2).optional().build(), new BigDecimal("1.10")),
      // DBZ-351 new SchemaAndValueField("dzs", Decimal.builder(0).optional().build(), new BigDecimal("10")),
```

Comment:
Seems this (and a few lines below) should be uncommented now, too?

Reply:
Thanks, I forgot about it